src/FactSystem/ParameterManager.cc

```diff
@@ -661,6 +661,8 @@ void ParameterManager::_setupDefaultComponentCategoryMap(void)
     for (const QString &paramName: _mapParameterName2Variant[_vehicle->defaultComponentId()].keys()) {
         Fact* fact = _mapParameterName2Variant[_vehicle->defaultComponentId()][paramName].value<Fact*>();
         defaultComponentCategoryMap[fact->category()][fact->group()] += paramName;
+        int i = paramName.indexOf("_");
+        defaultComponentCategoryMap[tr("All")][(i > 0) ? paramName.left(i) : tr("Misc")] += paramName;
```

Comment:
Are you explicitly not using group names here on purpose? I'm sort of 50/50 on whether that is good or bad. For PX4 the group names are way more human readable than just a name prefix.

Reply:
yes

although I can't really speak for PX4

I totally agree that the mavlink-ish naming should go and be replaced by human readable text. This should IMHO however so for both the groups and the parameter names. The Camera manager is a good example of there it should end up. 

However, that's not at all the current status (the COMPONENT_INFORMATION mechanism aims at getting there). The current state of affairs is in fact quite inconsistent, in that the parameters are still named by their mavlink names. This is so in QGC, also for PX4 (it is the first entry in the list and not the 3rd entry !!!). But this is so also in the docs !!! I.e., when you search for a parameter in the docs you have to search for the mavlink names, since these are given there (true for both PX4 and ArduPilot). Therefore, currently I think mavlink-ish sectioning is much more helpful. 

Once things are in a state there the mavlink names don't appear to the user, because only human readable names are used, both in QGC and the docs, I would agree with a human readable grouping also for a "All" category. However, although I certainly wish this state would be reached soon, experience shows that the timelines are long. Hence that choice.

Reply:
so, I myself am happy with this, I'm actually 50/50 with whether one should also for a "All" category take out eventual last digits from the group name, like you do in this other PR of yours. Split minded here. :)

Reply:
> whether one should also for a "All" category take out eventual last digits from the group name

Not quite sure what you mean there. 

But in thinking about this a bit I think All should use group names. Which should end up showing the same visual on ArduPilot but show humn readable group names for PX4.

Reply:
group.remove(QRegExp("[0-9]*$")); 

GUI design is subjective :)

Reply:
Ok, I get it now. I'd just leave it as is. My concern is that it will work and be good in most cases but may not be correct in all. The cases where it works it where the number indicates a series of parameters. But that may not always be the case for why a number is used. Imagine something like FOOV2_BAR.   Removing in that case is incorrect.

Reply:
I think it's just this last change and this can go in?

Reply:
thx for asking, much appreciated. This actually fell a bit off my mind, also since the simplification of removing "Other" you already did. I will rework it the next days.

Reply:
ok, looked at it. I think the last change you mentioned is missing is to show human readable group names for PX4. I unfortunately couldn't figure that out beyond doubt, and currently don't have a setup to test, so I'm sorry to say that I can't accomplish this task.

I hope to find time over Christmas break to get me into PX4, but for the moment that's the situation. Not sure if it's better to keep this PR open for a later starting point or to close it. Pl decide.

Reply:
> I unfortunately couldn't figure that out beyond doubt, and currently don't have a setup to test

You can use MockLinks to test things without requiring a real connection. App Settings/MockLink

Reply:
ah, very interesting comment. I'll see what I can make of it :) THX

Reply:
first, the MockLink is kinda cool, isn't it :D

I've played with this now, but there isn't no satisfying solution possible IMHO

I've changed my addition to `defaultComponentCategoryMap[tr("All")][fact->group()] += paramName;` which I think is what is supposed to be done.

the problem is the group 'Misc'. For PX4 there is a category 'Other' with group 'Misc' (to my surprise, I thought we killed the 'Other', so it must be a 'hardwired' thing), and for ArduPilot all unassigned parameters are put into a group 'Misc', which is found in the category 'Advanced'.

Therefore, with the above algorithm, there is also a group 'Misc' in the 'All' category, which is however not what one wants to have, since e.g. a parameter which starts with 'COM_' should be put into the group 'Commander' where all the other 'COM_' parameters are located. That is, the main purpose of having an 'All' category in the first place is defeated.

Now one could think that one could take that parameter from 'Misc' and try to find the proper group. In the example of the 'COM_' one could search for the group there the other 'COM_' params are located. However, this obviously only works unambiguously if either (i) any group has just parameters with the same header and/or (ii) if parameters with the same header are not distributed in several groups. However none of these conditions are fullfilled for PX4. There are several examples, the groups 'SensorCalibration' and 'Sensors' are most clear: In both one finds params with header 'CAL_' and 'SENS_', so it's impossible to sort out where a 'SENS_' or 'CAL_' parameter should go to. Ergo, no solution possible. Q.E.D.

A solution would only be possible if unassigned parameters could not ever occur, which obviously isn't fulfilled.

I have no idea how to proceed from here. :)